Jint/AstExtensions.cs

```diff
@@ -79,7 +79,16 @@ or NodeType.FunctionExpression
                 or NodeType.YieldExpression
                 or NodeType.TemplateLiteral)
             {
-                var context = engine._activeEvaluationContext;
+                var context = new EvaluationContext();
+
+                if (engine._activeEvaluationContext == null)
+                {
+                    context = new EvaluationContext(engine);
+                }
+                else
+                {
+                    context = engine._activeEvaluationContext;
+                }
```

Comment:
I've tried to write code this way, but it doesn't work. You can ensure if you run the test I provided. With improved version it will throw 'object is not a reference' error anyway. So, I'd left it as is